[Bug] Intermittent test failure gradle-dependencies: GIVEN strict policy WHEN some tracking policies THEN purge cookies

See also #15981 which shares the GC failures and was failed at the same time as this failed

Log: https://treeherder.mozilla.org/logviewer.html#?job_id=318890162&repo=fenix

```
[task 2020-10-16T23:24:32.548Z] SUITE: org.mozilla.fenix.components.TrackingProtectionPolicyFactoryTest
[task 2020-10-16T23:24:33.146Z]   TEST: GIVEN strict policy WHEN some tracking policies THEN purge cookies
[task 2020-10-16T23:24:33.146Z]     [Robolectric] org.mozilla.fenix.components.TrackingProtectionPolicyFactoryTest.GIVEN strict policy WHEN some tracking policies THEN purge cookies: sdk=28; resources=BINARY
[task 2020-10-16T23:24:33.146Z]     I/MonitoringInstr: Instrumentation started!
[task 2020-10-16T23:24:33.246Z]   FAILURE
[task 2020-10-16T23:24:33.246Z] 
[task 2020-10-16T23:24:33.246Z] java.lang.AssertionError: expected:<false> but was:<true>
[task 2020-10-16T23:24:33.246Z] 	at org.junit.Assert.fail(Assert.java:88)
[task 2020-10-16T23:24:33.246Z] 	at org.junit.Assert.failNotEquals(Assert.java:834)
[task 2020-10-16T23:24:33.246Z] 	at org.junit.Assert.assertEquals(Assert.java:118)
[task 2020-10-16T23:24:33.246Z] 	at org.junit.Assert.assertEquals(Assert.java:144)
[task 2020-10-16T23:24:33.246Z] 	at org.mozilla.fenix.components.TrackingProtectionPolicyFactoryTest.GIVEN strict policy WHEN some tracking policies THEN purge cookies(TrackingProtectionPolicyFactoryTest.kt:379)
[task 2020-10-16T23:24:33.246Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[task 2020-10-16T23:24:33.246Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[task 2020-10-16T23:24:33.246Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[task 2020-10-16T23:24:33.246Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[task 2020-10-16T23:24:33.246Z] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
[task 2020-10-16T23:24:33.246Z] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
[task 2020-10-16T23:24:33.246Z] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
[task 2020-10-16T23:24:33.246Z] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
[task 2020-10-16T23:24:33.246Z] 	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:546)
[task 2020-10-16T23:24:33.246Z] 	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:252)
[task 2020-10-16T23:24:33.246Z] 	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
[task 2020-10-16T23:24:33.246Z] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[task 2020-10-16T23:24:33.246Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[task 2020-10-16T23:24:33.246Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[task 2020-10-16T23:24:33.246Z] 	at java.lang.Thread.run(Thread.java:748)
```

--- a/app/src/test/java/org/mozilla/fenix/components/TrackingProtectionPolicyFactoryTest.kt
+++ b/app/src/test/java/org/mozilla/fenix/components/TrackingProtectionPolicyFactoryTest.kt
@@ -9,17 +9,24 @@ import org.junit.Assert.assertArrayEquals
 import org.junit.Assert.assertEquals
 import org.junit.Assert.assertFalse
 import org.junit.Assert.assertTrue
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.mozilla.fenix.Config
 import org.mozilla.fenix.FeatureFlags
 import org.mozilla.fenix.R
 import org.mozilla.fenix.ReleaseChannel
-import org.mozilla.fenix.utils.Settings
 import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+import org.mozilla.fenix.utils.Settings
 
 @RunWith(FenixRobolectricTestRunner::class)
 class TrackingProtectionPolicyFactoryTest {
+
+    @Before
+    fun setup() {
+        mockkObject(Config)
+        every { Config.channel } returns ReleaseChannel.Nightly
+    }
 
     @Test
     fun `WHEN useStrictMode is true then SHOULD return strict mode`() {

